libs/blocks/modal/modal.js

```diff
@@ -162,3 +164,15 @@ window.addEventListener('hashchange', (e) => {
     if (details) getModal(details);
   }
 });
+
+function sendViewportDimensionsToiFrame(source) {
+  const viewportWidth = Math.max(document.documentElement.clientWidth || 0, window.innerWidth || 0);
+  source.postMessage({ mobileMax: MOBILE_MAX, tabletMax: TABLET_MAX, viewportWidth }, '*');
+}
+
+window.addEventListener('message', (t) => {
+  if (t.data === 'viewportWidth') {
+    sendViewportDimensionsToiFrame(t.source);
+    window.addEventListener('resize', () => sendViewportDimensionsToiFrame(t.source));
```

Comment:
This should probably be debounced and it should be ensured that it's added just once.